src/main/scala/com/snowplowanalytics/indicative/LambdaHandler.scala

```diff
@@ -0,0 +1,43 @@
+package com.snowplowanalytics.indicative
+
+// Scala
+import scala.collection.JavaConverters._
+
+// cats
+import cats.implicits._
+
+// AWS
+import com.amazonaws.services.lambda.runtime.events.KinesisEvent
+
+// This library
+import com.snowplowanalytics.indicative.Transformer.TransformationError
+import com.snowplowanalytics.snowplow.analytics.scalasdk.json.EventTransformer
+
+class LambdaHandler {
+
+  val apiKey: String =
+    sys.env.getOrElse("INDICATIVE_API_KEY",
+                      throw new RuntimeException("You must provide environment variable INDICATIVE_API_KEY"))
+
+  def recordHandler(event: KinesisEvent): Unit = {
+    val events = event.getRecords.asScala.map { record =>
+      val line      = new String(record.getKinesis.getData.array, "UTF-8")
```

Comment:
are all those `get` safe? i.e. no npe can happen

Reply:
Yeah technically it can for `getRecords` and `getKinesis`, as there are empty constructors for these classes. However this is all handled by the AWS, do you think we can trust them on this one?

Reply:
never trust anyone, first rule of programming :smile:

Reply:
My understanding is that if these methods fail, it would be a programming error and there is nothing we can do to recover from this. What would you change here?

Reply:
As a first step I would Option-wrap them and then I'd look into the javadoc to see what the expected failure cases are